metricq/source.py

```diff
@@ -143,11 +138,21 @@ def task(self):
         """
         pass
 
-    def __getitem__(self, id):
+    def __getitem__(self, id: Metric) -> SourceMetric:
         if id not in self.metrics:
             self.metrics[id] = SourceMetric(id, self, chunk_size=self.chunk_size)
         return self.metrics[id]
 
+    def _augment_metadata(
+        self, metrics: Dict[Metric, MetadataDict]
+    ) -> Dict[Metric, MetadataDict]:
+        augmented: Dict[Metric, MetadataDict] = dict(**metrics)
+        for metric, metadata in augmented.items():
```

Comment:
Why is the `augmented` local variable needed? It looks strangely like the input `metrics`.

If for some reason a copy is needed, why not use `copy.copy()` ?

Reply:
I'll use `dict.copy()` and add a short comment.

TL;DR: from the outside, `metrics` seems to be a read-only object, so I don't want to have subvert a users expectations and modify the dictionary they own.